tests/publiccloud/transfer_repos.pm

```diff
@@ -32,7 +32,7 @@ sub run {
         record_info('Skip download', 'Skipping download triggered by setting (QAM_PUBLICCLOUD_SKIP_DOWNLOAD = 1)');
     } else {
         assert_script_run('du -sh ~/repos');
-        assert_script_run("rsync -uva -e ssh ~/repos root@" . $args->{my_instance}->public_ip . ":'/tmp/repos'", timeout => 1800);
+        assert_script_run("timeout 2400 rsync -uvaP -e ssh ~/repos root@" . $args->{my_instance}->public_ip . ":'/tmp/repos'", timeout => 2400);
```

Comment:
I would say that situation is the same as with curl progress , this will spam output quite a lot so I would keep it optional ( e.g. via some variable ) . We can even use same variable as for curl or add a separate one ...

also not sure what the point having timeout which duplicate functionality which we already have out of the box ( failing after defined period of time ) ?

Reply:
it can make sense to have internal timeouts as they can be more fine-grained. In this case I would say the bash-level timeout command is not adding benefits on top of the timeout in the os-autoinst level timeout. On top internal timeouts should always be lower than any higher level of timeout. rsync has an internal timeout parameter. How about:

```suggestion
        my $timeout = 2400;
        assert_script_run("rsync --timeout=$timeout -uvaP -e ssh ~/repos root@" . $args->{my_instance}->public_ip . ":'/tmp/repos'", timeout => $timeout + 10);
```

Reply:
@asmorodskyi the `rsync` currently prints every file it transfers. The `-P` arguments only adds additional data there. This patch updates one specific test module - I don't think we need special variable for that.

@okurz right now, when the `assert_script_run` times out, the `curl` still runs which makes the terminal unusable. Thank you for the internal timeout parameter.

Reply:
ok agree , and I like what Oli suggesting

Reply:
Me too, I already implemented it. Now running verification runs.